core/src/main/java/org/sonarsource/sonarlint/core/embedded/server/ShowIssueRequestHandler.java

```diff
@@ -185,19 +187,24 @@ static ShowIssueQuery extractQuery(ClassicHttpRequest request) {
     } catch (URISyntaxException e) {
       // Ignored
     }
-    return new ShowIssueQuery(params.get("server"), params.get("project"), params.get("issue"));
+    return new ShowIssueQuery(params.get("server"), params.get("project"), params.get("issue"), params.get("branch"), params.get("pullRequest"));
   }
 
   @VisibleForTesting
   public static class ShowIssueQuery {
     private final String serverUrl;
     private final String projectKey;
     private final String issueKey;
+    private final String branch;
+    @Nullable
+    private final String pullRequest;
 
-    public ShowIssueQuery(String serverUrl, String projectKey, String issueKey) {
+    public ShowIssueQuery(String serverUrl, String projectKey, String issueKey, String branch, @Nullable String pullRequest) {
       this.serverUrl = serverUrl;
       this.projectKey = projectKey;
       this.issueKey = issueKey;
+      this.branch = branch;
+      this.pullRequest = pullRequest;
     }
 
     public boolean isValid() {
```

Comment:
You might want to change isValid to check if you receive either a branch or a PR? Not sure about the exact check